frontend/src/Context/SessionProvider/session.tsx

```diff
@@ -368,51 +237,67 @@ const SessionProvider = ({ children }: SessionProviderProps): ReactElement => {
+    setSubscriberWrappers((previousSubscriberWrappers) =>
+      [subscriberWrapper, ...previousSubscriberWrappers].sort(
+        sortByDisplayPriority(activeSpeakerIdRef.current)
+      )
+    );
+  };
+
+  const handleSubscriberDestroyed = (streamId: string) => {
+    activeSpeakerTracker.current.onSubscriberDestroyed(streamId);
+    const isNotDestroyedStreamId = ({ id }: { id: string }) => streamId !== id;
+    setSubscriberWrappers((prevSubscriberWrappers) =>
+      prevSubscriberWrappers.filter(isNotDestroyedStreamId)
+    );
+  };
+
+  const handleSubscriberAudioLevelUpdated = ({
+    movingAvg,
+    subscriberId,
+  }: SubscriberAudioLevelUpdatedEvent) => {
+    activeSpeakerTracker.current.onSubscriberAudioLevelUpdated({
+      subscriberId,
+      movingAvg,
+    });
+  };
+
   /**
    * Connects to the session using the provided credentials.
    * @param {Credential} credential - The credentials for the session.
    * @returns {Promise<void>} A promise that resolves when the session is connected.
    */
   const connect = useCallback(async (credential: Credential) => {
+    if (session.current) {
+      return;
+    }
     try {
-      const { apiKey, sessionId, token } = credential;
       // initialize the session object and set up the relevant event listeners
       // https://tokbox.com/developer/sdks/js/reference/Session.html#events for opentok
       // https://vonage.github.io/conversation-docs/video-js-reference/latest/Session.html#events for unified environment
-      session.current = initSession(apiKey, sessionId);
-
+      session.current = new VonageVideoClient(credential);
       session.current.on('streamPropertyChanged', handleStreamPropertyChanged);
-      session.current.on('streamCreated', handleStreamCreated);
       session.current.on('sessionReconnecting', handleReconnecting);
       session.current.on('sessionReconnected', handleReconnected);
       session.current.on('sessionDisconnected', handleSessionDisconnected);
-      session.current.on('connectionCreated', handleConnectionCreated);
-      session.current.on('connectionDestroyed', handleConnectionDestroyed);
       session.current.on('archiveStarted', handleArchiveStarted);
       session.current.on('archiveStopped', handleArchiveStopped);
-      // @ts-expect-error signal:<type> is not ts compliant
       session.current.on('signal:chat', handleChatSignal);
-
-      await new Promise<string | undefined>((resolve, reject) => {
-        session.current?.connect(token, (err?: OTError) => {
-          if (err) {
-            // We ignore the following lint warning because we are rejecting with an OTError object.
-            reject(err); // NOSONAR
-          } else {
-            setConnected(true);
-            logOnConnect(apiKey, sessionId, session.current?.connection?.connectionId);
-            resolve(session.current?.sessionId);
-          }
-        });
-      });
```

Comment:
Moved to `VonageVideoClient` file. We set `connected` state if we successfully connect, below.

Reply:
can this error though?

Reply:
Yep! The `session.current.connect(credential);` call can throw if there's an issue calling `Session.connect`